### ISSUE TITLE
Gosec question

### ISSUE THREAD

**Issue by szombi** [oauth2-proxy/oauth2-proxy#692]
Question about gosec and one example report 

- Version used: 5.1.0

Hello! We are currently using 5.1.0 oauth2-proxy in our product and based on our security rules we ran gosec on your code. I attached an example security report (most of can be false positive but I don't know your code). My question is do you plan to use gosec during the build or before releases?  

Thank you!

```
[gosec] 2020/07/20 10:16:40 Including rules: default
[gosec] 2020/07/20 10:16:40 Excluding rules: default
[gosec] 2020/07/20 10:16:40 Import directory: /go/src/github.com/oauth2-proxy/pkg/validation
[gosec] 2020/07/20 10:16:48 Checking package: validation
[gosec] 2020/07/20 10:16:48 Checking file: /go/src/github.com/oauth2-proxy/pkg/validation/cookie.go
[gosec] 2020/07/20 10:16:48 Checking file: /go/src/github.com/oauth2-proxy/pkg/validation/logging.go
[gosec] 2020/07/20 10:16:48 Checking file: /go/src/github.com/oauth2-proxy/pkg/validation/options.go
[gosec] 2020/07/20 10:16:48 Checking file: /go/src/github.com/oauth2-proxy/pkg/validation/sessions.go
[gosec] 2020/07/20 10:16:48 Checking file: /go/src/github.com/oauth2-proxy/pkg/validation/upstreams.go
[gosec] 2020/07/20 10:16:48 Import directory: /go/src/github.com/oauth2-proxy/pkg/requests
[gosec] 2020/07/20 10:16:48 Checking package: requests
[gosec] 2020/07/20 10:16:48 Checking file: /go/src/github.com/oauth2-proxy/pkg/requests/builder.go
[gosec] 2020/07/20 10:16:48 Checking file: /go/src/github.com/oauth2-proxy/pkg/requests/result.go
[gosec] 2020/07/20 10:16:48 Import directory: /go/src/github.com/oauth2-proxy/pkg/sessions/cookie
[gosec] 2020/07/20 10:16:48 Checking package: cookie
[gosec] 2020/07/20 10:16:48 Checking file: /go/src/github.com/oauth2-proxy/pkg/sessions/cookie/session_store.go
[gosec] 2020/07/20 10:16:48 Import directory: /go/src/github.com/oauth2-proxy/pkg/sessions/persistence
[gosec] 2020/07/20 10:16:49 Checking package: persistence
[gosec] 2020/07/20 10:16:49 Checking file: /go/src/github.com/oauth2-proxy/pkg/sessions/persistence/interfaces.go
[gosec] 2020/07/20 10:16:49 Checking file: /go/src/github.com/oauth2-proxy/pkg/sessions/persistence/manager.go
[gosec] 2020/07/20 10:16:49 Checking file: /go/src/github.com/oauth2-proxy/pkg/sessions/persistence/ticket.go
[gosec] 2020/07/20 10:16:49 Import directory: /go/src/github.com/oauth2-proxy/pkg/sessions/redis
[gosec] 2020/07/20 10:16:51 Checking package: redis
[gosec] 2020/07/20 10:16:51 Checking file: /go/src/github.com/oauth2-proxy/pkg/sessions/redis/client.go
[gosec] 2020/07/20 10:16:51 Checking file: /go/src/github.com/oauth2-proxy/pkg/sessions/redis/redis_store.go
[gosec] 2020/07/20 10:16:51 Import directory: /go/src/github.com/oauth2-proxy/pkg/util
[gosec] 2020/07/20 10:16:51 Checking package: util
[gosec] 2020/07/20 10:16:51 Checking file: /go/src/github.com/oauth2-proxy/pkg/util/util.go
[gosec] 2020/07/20 10:16:51 Import directory: /go/src/github.com/oauth2-proxy/pkg/apis/middleware
[gosec] 2020/07/20 10:16:51 Checking package: middleware
[gosec] 2020/07/20 10:16:51 Checking file: /go/src/github.com/oauth2-proxy/pkg/apis/middleware/scope.go
[gosec] 2020/07/20 10:16:51 Checking file: /go/src/github.com/oauth2-proxy/pkg/apis/middleware/session.go
[gosec] 2020/07/20 10:16:51 Import directory: /go/src/github.com/oauth2-proxy/pkg/logger
[gosec] 2020/07/20 10:16:51 Checking package: logger
[gosec] 2020/07/20 10:16:51 Checking file: /go/src/github.com/oauth2-proxy/pkg/logger/logger.go
[gosec] 2020/07/20 10:16:51 Import directory: /go/src/github.com/oauth2-proxy/pkg/sessions
[gosec] 2020/07/20 10:16:52 Checking package: sessions
[gosec] 2020/07/20 10:16:52 Checking file: /go/src/github.com/oauth2-proxy/pkg/sessions/session_store.go
[gosec] 2020/07/20 10:16:52 Import directory: /go/src/github.com/oauth2-proxy/providers
[gosec] 2020/07/20 10:16:53 Checking package: providers
[gosec] 2020/07/20 10:16:53 Checking file: /go/src/github.com/oauth2-proxy/providers/azure.go
[gosec] 2020/07/20 10:16:53 Checking file: /go/src/github.com/oauth2-proxy/providers/bitbucket.go
[gosec] 2020/07/20 10:16:53 Checking file: /go/src/github.com/oauth2-proxy/providers/digitalocean.go
[gosec] 2020/07/20 10:16:53 Checking file: /go/src/github.com/oauth2-proxy/providers/facebook.go
[gosec] 2020/07/20 10:16:53 Checking file: /go/src/github.com/oauth2-proxy/providers/github.go
[gosec] 2020/07/20 10:16:53 Checking file: /go/src/github.com/oauth2-proxy/providers/gitlab.go
[gosec] 2020/07/20 10:16:53 Checking file: /go/src/github.com/oauth2-proxy/providers/google.go
[gosec] 2020/07/20 10:16:53 Checking file: /go/src/github.com/oauth2-proxy/providers/internal_util.go
[gosec] 2020/07/20 10:16:53 Checking file: /go/src/github.com/oauth2-proxy/providers/keycloak.go
[gosec] 2020/07/20 10:16:53 Checking file: /go/src/github.com/oauth2-proxy/providers/linkedin.go
[gosec] 2020/07/20 10:16:53 Checking file: /go/src/github.com/oauth2-proxy/providers/logingov.go
[gosec] 2020/07/20 10:16:53 Checking file: /go/src/github.com/oauth2-proxy/providers/nextcloud.go
[gosec] 2020/07/20 10:16:53 Checking file: /go/src/github.com/oauth2-proxy/providers/oidc.go
[gosec] 2020/07/20 10:16:53 Checking file: /go/src/github.com/oauth2-proxy/providers/provider_data.go
[gosec] 2020/07/20 10:16:53 Checking file: /go/src/github.com/oauth2-proxy/providers/provider_default.go
[gosec] 2020/07/20 10:16:53 Checking file: /go/src/github.com/oauth2-proxy/providers/providers.go
[gosec] 2020/07/20 10:16:53 Import directory: /go/src/github.com/oauth2-proxy
[gosec] 2020/07/20 10:16:54 Checking package: main
[gosec] 2020/07/20 10:16:54 Checking file: /go/src/github.com/oauth2-proxy/http.go
[gosec] 2020/07/20 10:16:54 Checking file: /go/src/github.com/oauth2-proxy/logging_handler.go
[gosec] 2020/07/20 10:16:54 Checking file: /go/src/github.com/oauth2-proxy/main.go
[gosec] 2020/07/20 10:16:54 Checking file: /go/src/github.com/oauth2-proxy/oauthproxy.go
[gosec] 2020/07/20 10:16:54 Checking file: /go/src/github.com/oauth2-proxy/templates.go
[gosec] 2020/07/20 10:16:54 Checking file: /go/src/github.com/oauth2-proxy/validator.go
[gosec] 2020/07/20 10:16:54 Checking file: /go/src/github.com/oauth2-proxy/version.go
[gosec] 2020/07/20 10:16:54 Checking file: /go/src/github.com/oauth2-proxy/watcher.go
[gosec] 2020/07/20 10:16:54 Import directory: /go/src/github.com/oauth2-proxy/pkg/apis/ip
[gosec] 2020/07/20 10:16:54 Checking package: ip
[gosec] 2020/07/20 10:16:54 Checking file: /go/src/github.com/oauth2-proxy/pkg/apis/ip/interfaces.go
[gosec] 2020/07/20 10:16:54 Import directory: /go/src/github.com/oauth2-proxy/pkg/apis/options
[gosec] 2020/07/20 10:16:54 Checking package: options
[gosec] 2020/07/20 10:16:54 Checking file: /go/src/github.com/oauth2-proxy/pkg/apis/options/cookie.go
[gosec] 2020/07/20 10:16:54 Checking file: /go/src/github.com/oauth2-proxy/pkg/apis/options/legacy_options.go
[gosec] 2020/07/20 10:16:54 Checking file: /go/src/github.com/oauth2-proxy/pkg/apis/options/load.go
[gosec] 2020/07/20 10:16:54 Checking file: /go/src/github.com/oauth2-proxy/pkg/apis/options/logging.go
[gosec] 2020/07/20 10:16:54 Checking file: /go/src/github.com/oauth2-proxy/pkg/apis/options/options.go
[gosec] 2020/07/20 10:16:54 Checking file: /go/src/github.com/oauth2-proxy/pkg/apis/options/sessions.go
[gosec] 2020/07/20 10:16:54 Checking file: /go/src/github.com/oauth2-proxy/pkg/apis/options/upstreams.go
[gosec] 2020/07/20 10:16:54 Import directory: /go/src/github.com/oauth2-proxy/pkg/cookies
[gosec] 2020/07/20 10:16:55 Checking package: cookies
[gosec] 2020/07/20 10:16:55 Checking file: /go/src/github.com/oauth2-proxy/pkg/cookies/cookies.go
[gosec] 2020/07/20 10:16:55 Import directory: /go/src/github.com/oauth2-proxy/pkg/ip
[gosec] 2020/07/20 10:16:55 Checking package: ip
[gosec] 2020/07/20 10:16:55 Checking file: /go/src/github.com/oauth2-proxy/pkg/ip/net_set.go
[gosec] 2020/07/20 10:16:55 Checking file: /go/src/github.com/oauth2-proxy/pkg/ip/parse_ip_net.go
[gosec] 2020/07/20 10:16:55 Checking file: /go/src/github.com/oauth2-proxy/pkg/ip/realclientip.go
[gosec] 2020/07/20 10:16:55 Import directory: /go/src/github.com/oauth2-proxy/pkg/upstream
[gosec] 2020/07/20 10:16:56 Checking package: upstream
[gosec] 2020/07/20 10:16:56 Checking file: /go/src/github.com/oauth2-proxy/pkg/upstream/file.go
[gosec] 2020/07/20 10:16:56 Checking file: /go/src/github.com/oauth2-proxy/pkg/upstream/http.go
[gosec] 2020/07/20 10:16:56 Checking file: /go/src/github.com/oauth2-proxy/pkg/upstream/proxy.go
[gosec] 2020/07/20 10:16:56 Checking file: /go/src/github.com/oauth2-proxy/pkg/upstream/static.go
[gosec] 2020/07/20 10:16:56 Import directory: /go/src/github.com/oauth2-proxy/pkg/apis/sessions
[gosec] 2020/07/20 10:16:56 Checking package: sessions
[gosec] 2020/07/20 10:16:56 Checking file: /go/src/github.com/oauth2-proxy/pkg/apis/sessions/interfaces.go
[gosec] 2020/07/20 10:16:56 Checking file: /go/src/github.com/oauth2-proxy/pkg/apis/sessions/legacy_v5_tester.go
[gosec] 2020/07/20 10:16:56 Checking file: /go/src/github.com/oauth2-proxy/pkg/apis/sessions/session_state.go
[gosec] 2020/07/20 10:16:56 Import directory: /go/src/github.com/oauth2-proxy/pkg/authentication/basic
[gosec] 2020/07/20 10:16:56 Checking package: basic
[gosec] 2020/07/20 10:16:56 Checking file: /go/src/github.com/oauth2-proxy/pkg/authentication/basic/htpasswd.go
[gosec] 2020/07/20 10:16:56 Checking file: /go/src/github.com/oauth2-proxy/pkg/authentication/basic/validator.go
[gosec] 2020/07/20 10:16:56 Import directory: /go/src/github.com/oauth2-proxy/pkg/encryption
[gosec] 2020/07/20 10:16:56 Checking package: encryption
[gosec] 2020/07/20 10:16:56 Checking file: /go/src/github.com/oauth2-proxy/pkg/encryption/cipher.go
[gosec] 2020/07/20 10:16:56 Checking file: /go/src/github.com/oauth2-proxy/pkg/encryption/nonce.go
[gosec] 2020/07/20 10:16:56 Checking file: /go/src/github.com/oauth2-proxy/pkg/encryption/utils.go
[gosec] 2020/07/20 10:16:56 Import directory: /go/src/github.com/oauth2-proxy/pkg/middleware
[gosec] 2020/07/20 10:16:57 Checking package: middleware
[gosec] 2020/07/20 10:16:57 Checking file: /go/src/github.com/oauth2-proxy/pkg/middleware/basic_session.go
[gosec] 2020/07/20 10:16:57 Checking file: /go/src/github.com/oauth2-proxy/pkg/middleware/healthcheck.go
[gosec] 2020/07/20 10:16:57 Checking file: /go/src/github.com/oauth2-proxy/pkg/middleware/jwt_session.go
[gosec] 2020/07/20 10:16:57 Checking file: /go/src/github.com/oauth2-proxy/pkg/middleware/redirect_to_https.go
[gosec] 2020/07/20 10:16:57 Checking file: /go/src/github.com/oauth2-proxy/pkg/middleware/scope.go
[gosec] 2020/07/20 10:16:57 Checking file: /go/src/github.com/oauth2-proxy/pkg/middleware/session_utils.go
[gosec] 2020/07/20 10:16:57 Checking file: /go/src/github.com/oauth2-proxy/pkg/middleware/stored_session.go
[gosec] 2020/07/20 10:16:57 Import directory: /go/src/github.com/oauth2-proxy/pkg/sessions/tests
[gosec] 2020/07/20 10:16:58 Checking package: tests
[gosec] 2020/07/20 10:16:58 Checking file: /go/src/github.com/oauth2-proxy/pkg/sessions/tests/mock_store.go
[gosec] 2020/07/20 10:16:58 Checking file: /go/src/github.com/oauth2-proxy/pkg/sessions/tests/session_store_tests.go
Results:

Golang errors in file: [/go/src/github.com/oauth2-proxy/oauthproxy.go]:

  > [line 979 : column 29] - req.Header.Values undefined (type http.Header has no field or method Values)



[/go/src/github.com/oauth2-proxy/pkg/upstream/http.go:160] - G402 (CWE-295): TLS InsecureSkipVerify set true. (Confidence: HIGH, Severity: HIGH)
  > InsecureSkipVerify: true


[/go/src/github.com/oauth2-proxy/pkg/middleware/jwt_session.go:124] - G101 (CWE-798): Potential hardcoded credentials (Confidence: LOW, Severity: HIGH)
  > password == "x-oauth-basic"


[/go/src/github.com/oauth2-proxy/pkg/validation/options.go:34] - G402 (CWE-295): TLS InsecureSkipVerify set true. (Confidence: HIGH, Severity: HIGH)
  > InsecureSkipVerify: true


[/go/src/github.com/oauth2-proxy/pkg/upstream/http.go:108] - G402 (CWE-295): TLS InsecureSkipVerify set true. (Confidence: HIGH, Severity: HIGH)
  > InsecureSkipVerify: true


[/go/src/github.com/oauth2-proxy/oauthproxy.go:412] - G203 (CWE-79): this method will not auto-escape HTML. Verify data is well formed. (Confidence: LOW, Severity: MEDIUM)
  > template.HTML(p.SignInMessage)


[/go/src/github.com/oauth2-proxy/pkg/authentication/basic/htpasswd.go:88] - G401 (CWE-326): Use of weak cryptographic primitive (Confidence: HIGH, Severity: MEDIUM)
  > sha1.New()


[/go/src/github.com/oauth2-proxy/pkg/authentication/basic/htpasswd.go:36] - G307 (CWE-): Deferring unsafe method "*os.File" on type "Close" (Confidence: HIGH, Severity: MEDIUM)
  > defer r.Close()


[/go/src/github.com/oauth2-proxy/pkg/authentication/basic/htpasswd.go:32] - G304 (CWE-22): Potential file inclusion via variable (Confidence: HIGH, Severity: MEDIUM)
  > os.Open(path)


[/go/src/github.com/oauth2-proxy/pkg/authentication/basic/htpasswd.go:4] - G505 (CWE-327): Blacklisted import crypto/sha1: weak cryptographic primitive (Confidence: HIGH, Severity: MEDIUM)
  > "crypto/sha1"


[/go/src/github.com/oauth2-proxy/pkg/encryption/utils.go:5] - G505 (CWE-327): Blacklisted import crypto/sha1: weak cryptographic primitive (Confidence: HIGH, Severity: MEDIUM)
  > "crypto/sha1"


[/go/src/github.com/oauth2-proxy/pkg/util/util.go:15] - G304 (CWE-22): Potential file inclusion via variable (Confidence: HIGH, Severity: MEDIUM)
  > ioutil.ReadFile(path)


[/go/src/github.com/oauth2-proxy/validator.go:50] - G307 (CWE-): Deferring unsafe method "*os.File" on type "Close" (Confidence: HIGH, Severity: MEDIUM)
  > defer r.Close()


[/go/src/github.com/oauth2-proxy/pkg/validation/logging.go:16] - G302 (CWE-276): Expect file permissions to be 0600 or less (Confidence: HIGH, Severity: MEDIUM)
  > os.OpenFile(o.File.Filename, os.O_WRONLY|os.O_CREATE, 0666)


[/go/src/github.com/oauth2-proxy/oauthproxy.go:417] - G203 (CWE-79): this method will not auto-escape HTML. Verify data is well formed. (Confidence: LOW, Severity: MEDIUM)
  > template.HTML(p.Footer)


[/go/src/github.com/oauth2-proxy/oauthproxy.go:637] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > p.SaveSession(rw, req, session)


[/go/src/github.com/oauth2-proxy/oauthproxy.go:388] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > p.ClearSessionCookie(rw, req)


[/go/src/github.com/oauth2-proxy/oauthproxy.go:382] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > p.templates.ExecuteTemplate(rw, "error.html", t)


[/go/src/github.com/oauth2-proxy/main.go:28] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > flagSet.Parse(os.Args[1:])


[/go/src/github.com/oauth2-proxy/oauthproxy.go:422] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > p.templates.ExecuteTemplate(rw, "sign_in.html", t)


[/go/src/github.com/oauth2-proxy/http.go:128] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > tc.SetKeepAlivePeriod(3 * time.Minute)


[/go/src/github.com/oauth2-proxy/oauthproxy.go:665] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > json.NewEncoder(rw).Encode(userInfo)


[/go/src/github.com/oauth2-proxy/oauthproxy.go:676] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > p.ClearSessionCookie(rw, req)


[/go/src/github.com/oauth2-proxy/oauthproxy.go:839] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > p.ClearSessionCookie(rw, req)


[/go/src/github.com/oauth2-proxy/validator.go:24] - G103 (CWE-242): Use of unsafe calls should be audited (Confidence: HIGH, Severity: LOW)
  > unsafe.Pointer(&m)


[/go/src/github.com/oauth2-proxy/http.go:127] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > tc.SetKeepAlive(true)


[/go/src/github.com/oauth2-proxy/validator.go:65] - G103 (CWE-242): Use of unsafe calls should be audited (Confidence: HIGH, Severity: LOW)
  > unsafe.Pointer(&updated)


[/go/src/github.com/oauth2-proxy/watcher.go:58] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > watcher.Remove(filename)


[/go/src/github.com/oauth2-proxy/providers/logingov.go:106] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > resp.Body.Close()


[/go/src/github.com/oauth2-proxy/pkg/logger/logger.go:226] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > l.writer.Write([]byte("\n"))


[/go/src/github.com/oauth2-proxy/pkg/upstream/proxy.go:88] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > errorTemplate.Execute(rw, data)


[/go/src/github.com/oauth2-proxy/pkg/logger/logger.go:211-224] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > l.reqTemplate.Execute(l.writer, reqLogMessageData{
                Client:          client,
                Host:            req.Host,
                Protocol:        req.Proto,
                RequestDuration: fmt.Sprintf("%0.3f", duration),
                RequestMethod:   req.Method,
                RequestURI:      fmt.Sprintf("%q", url.RequestURI()),
                ResponseSize:    fmt.Sprintf("%d", size),
                StatusCode:      fmt.Sprintf("%d", status),
                Timestamp:       FormatTimestamp(ts),
                Upstream:        upstream,
                UserAgent:       fmt.Sprintf("%q", req.UserAgent()),
                Username:        username,
        })


[/go/src/github.com/oauth2-proxy/pkg/logger/logger.go:175] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > l.writer.Write([]byte("\n"))


[/go/src/github.com/oauth2-proxy/pkg/logger/logger.go:163-173] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > l.authTemplate.Execute(l.writer, authLogMessageData{
                Client:        client,
                Host:          req.Host,
                Protocol:      req.Proto,
                RequestMethod: req.Method,
                Timestamp:     FormatTimestamp(now),
                UserAgent:     fmt.Sprintf("%q", req.UserAgent()),
                Username:      username,
                Status:        string(status),
                Message:       fmt.Sprintf(format, a...),
        })


[/go/src/github.com/oauth2-proxy/pkg/logger/logger.go:141] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > l.writer.Write([]byte("\n"))


[/go/src/github.com/oauth2-proxy/pkg/authentication/basic/htpasswd.go:89] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > d.Write([]byte(password))


[/go/src/github.com/oauth2-proxy/pkg/logger/logger.go:135-139] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > l.stdLogTemplate.Execute(l.writer, stdLogMessageData{
                Timestamp: FormatTimestamp(now),
                File:      file,
                Message:   message,
        })


[/go/src/github.com/oauth2-proxy/pkg/encryption/utils.go:79] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > h.Write([]byte(arg))


[/go/src/github.com/oauth2-proxy/pkg/validation/logging.go:22] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > file.Close()


[/go/src/github.com/oauth2-proxy/pkg/middleware/stored_session.go:76] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > s.store.Clear(rw, req)


Summary:
   Files: 75
   Lines: 9775
   Nosec: 0
  Issues: 39

```

**Comment by JoelSpeed** [comment]
What perfect timing, #690 was just opened to resolve a number of these issues! 

> My question is do you plan to use gosec during the build or before releases?

I think that would be a great idea once we have it passing, I suspect that may be the case after #690, @NickMeves WDYT?

**Comment by NickMeves** [comment]
Well that list looks familiar 😉 

My vision is once we either clean up everything or mark it as `#nosec` we force a clean gosec on future PRs as a mergeability branch protection.

**Comment by szombi** [comment]
This sounds great! Thanks for the info! :)

**Comment by JoelSpeed** [comment]
Can we add `gosec` to the `golangci-lint` list? It's supported so maybe we just add that onto your PR @NickMeves?

**Comment by NickMeves** [comment]
Added it in #690 and confirmed it failed when I left a few gosec findings unmitigated to test.

That PR now has all of these findings mitigates or documented as to why we won't mitigate.

**Comment by NickMeves** [comment]
Resolved this here: https://github.com/oauth2-proxy/oauth2-proxy/pull/690

This was released in v6.1.0 last week.
